pkg/discovery/stitching/volume_stitching_manager.go

```diff
@@ -182,3 +184,26 @@ func (azure *azureVolumeUUIDGetter) GetVolumeUUID(vol *api.PersistentVolume) (st
 
 	return stitchingUUID, nil
 }
+
+type vsphereVolumeUUIDGetter struct {
+}
+
+func (azure *vsphereVolumeUUIDGetter) Name() string {
+	return "VSPHERE"
+}
+
+func (azure *vsphereVolumeUUIDGetter) GetVolumeUUID(vol *api.PersistentVolume) (string, error) {
+	if vol.Spec.VsphereVolume == nil {
+		return "", fmt.Errorf("not a valid Vsphere provisioned volume: %v", vol.Name)
+	}
+
+	path := vol.Spec.VsphereVolume.VolumePath
+	//1. Get the stitching id by transforming:
+	// [PUREM10:DS01] kubevols/kubernetes-dynamic-pvc-b9a45f70-0817-460c-a60f-9dac3c4b3d3f.vmdk
+	// to:
+	// PUREM10:DS01-kubevols-kubernetes-dynamic-pvc-b9a45f70-0817-48A1ADF83BA7B4A609DA
```

Comment:
Please put a ref to the rbcommons link so that we can easily track if anything changed.